src/_pytest/cacheprovider.py

```diff
@@ -289,6 +289,7 @@ def sort_key(node: nodes.Item | nodes.Collector) -> bool:
                     x
                     for x in result
                     if x.nodeid in lastfailed
+                    or x.nodeid.split("::")[0] in lastfailed
```

Comment:
Instead of sring manipulation this shoul det the nodeid of the File subclass parent